src/main/java/org/prebid/server/validation/RequestValidator.java

```diff
@@ -613,18 +619,23 @@ private void validateNativePlacementType(Integer placementType, int index) throw
         for (int i = 0; i < assets.size(); i++) {
             final Asset asset = assets.get(i);
             validateNativeAsset(asset, impIndex, i);
-            updatedAssets.add(asset.toBuilder().id(i).build());
+
+            final Asset updatedAsset = asset.getId() != null ? asset : asset.toBuilder().id(i).build();
+            final boolean hasAssetWithId = updatedAssets.stream()
+                    .map(Asset::getId)
+                    .anyMatch(id -> id.equals(updatedAsset.getId()));
+
+            if (hasAssetWithId) {
+                throw new ValidationException("request.imp[%d].native.request.assets[%d].id is already being used by "
+                        + "another asset. Each asset ID must be unique.", impIndex, i);
+            }
+
+            updatedAssets.add(updatedAsset);
         }
         return updatedAssets;
     }
 
     private void validateNativeAsset(Asset asset, int impIndex, int assetIndex) throws ValidationException {
-        if (asset.getId() != null) {
-            throw new ValidationException("request.imp[%d].native.request.assets[%d].id must not be defined. Prebid"
-                    + " Server will set this automatically, using the index of the asset in the array as the ID",
-                    impIndex, assetIndex);
-        }
-
```

Comment:
I know that's not your code, but could you also think about changing the 
line 633
`final long assetsCount = Stream.of(title, image, video, data)` 
to
`ObjectUtils.anyNotNull(....)`

Reply:
We actually need a stream to count the amount, so if there's more than one we throw validation exception.